### ISSUE TITLE
Investigate better Popup logic

### ISSUE THREAD

**Issue by rubengees** [vanniktech/Emoji#96]
- Version of the library: 0.5
- Affected devices: All
- Affected versions: All

The current logic for showing the picker itself and the skin tone popup have some problems.
This includes:
- Does not work with physical keyboards, because the `OnGlobalLayoutListener` does not get called (#31).
- Problems with dialogs and other contexts, apart from a normal full screen layout (#26).
- Skin tone popup does not behave exactly like it should be. If you do a long click, but dont lift your finger, you are still able to move the list. Moreover does an outside click counts as a real click. If you click outside the popup on an emoji, the popup is closed **and** the emoji is selected, while it should only close the popup.
- The picker does also not behave exactly like it should be. Other pickers like this close when clicked outside, but it is arguable if that is what we want.
- The popup closes when the `Activity` is paused, or orientation is changed. Other pickers like this do remain open
- Risky code
  - The `MIN_KEYBOARD_HEIGHT` is hardcoded to 100 pixels. I doubt that there are keyboards smaller than that, but it is still risky.
  - Usage of the Android internal field `status_bar_height`, which could change or disappear in future versions.

I tested around a bit, but couldn't really get to anything. One notable thing though is the following:
To make a popup window behave properly (Like I described above for the skin tone popup), one can set:

```java
popupWindow.setFocusable(true);
popupWindow.setOutsideTouchable(true);
```

This resulted in this weird behaviour though:

![unbenannt](https://cloud.githubusercontent.com/assets/8021265/23287097/987bba7c-fa3b-11e6-9444-a87474290008.png)

I don't really know how to approach this. Do you know something @vanniktech? I will further investigate this though.

**Comment by vanniktech** [comment]
> The picker does also not behave exactly like it should be. Other pickers like this close when clicked outside, but it is arguable if that is what we want.

I think the current behavior is what we want. 

> The popup closes when the Activity is paused, or orientation is changed. Other pickers like this do remain open

That might be the case. In 99% of my apps I only support portrait mode and do not really care about this behavior that much.

> The MIN_KEYBOARD_HEIGHT is hardcoded to 100 pixels. I doubt that there are keyboards smaller than that, but it is still risky.

Yup but that has always been the workaround for detecting keyboards and was The Solution™

> Usage of the Android internal field status_bar_height, which could change or disappear in future versions.

Also for that I'm not aware of another solution.

```java
popupWindow.setFocusable(true);
popupWindow.setOutsideTouchable(true);
```

Makes sense on first glance but there can be more corner cases to this other than the one you already saw.

I know that the implementation of the Popup is not the best however it works for most of the use cases.

**Comment by vanniktech** [comment]
Closed with #97. Thanks 👍
